src/main/java/org/opensearch/neuralsearch/search/collector/HybridCollapsingTopDocsCollector.java

```diff
@@ -0,0 +1,406 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.opensearch.neuralsearch.search.collector;
+
+import lombok.Setter;
+import lombok.extern.log4j.Log4j2;
+import org.apache.lucene.index.LeafReaderContext;
+import org.apache.lucene.search.CollectionTerminatedException;
+import org.apache.lucene.search.Collector;
+import org.apache.lucene.search.FieldComparator;
+import org.apache.lucene.search.FieldDoc;
+import org.apache.lucene.search.FieldValueHitQueue;
+import org.apache.lucene.search.LeafCollector;
+import org.apache.lucene.search.LeafFieldComparator;
+import org.apache.lucene.search.ScoreDoc;
+import org.apache.lucene.search.ScoreMode;
+import org.apache.lucene.search.Sort;
+import org.apache.lucene.search.SortField;
+import org.apache.lucene.search.TotalHits;
+import org.apache.lucene.search.grouping.CollapseTopFieldDocs;
+import org.apache.lucene.search.grouping.GroupSelector;
+import org.apache.lucene.util.BytesRef;
+import org.opensearch.index.mapper.MappedFieldType;
+import org.opensearch.neuralsearch.query.HybridSubQueryScorer;
+import org.opensearch.neuralsearch.search.HitsThresholdChecker;
+import org.opensearch.neuralsearch.search.lucene.MultiLeafFieldComparator;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Collects the CollapseTopFieldDocs based on a collapse field passed in a search request containing a hybrid query.
+ */
+@Log4j2
+public class HybridCollapsingTopDocsCollector<T> implements HybridSearchCollector, Collector {
+    protected final String collapseField;
+    private int totalHitCount;
+    private float maxScore = 0.0f;
+    private Sort sort;
+    private final GroupSelector<T> groupSelector;
+    private final int[] reversed;
+    private final boolean needsScores = true;
+    private int docBase;
+    private final HashMap<T, FieldValueHitQueue<FieldValueHitQueue.Entry>[]> groupQueueMap;
```

Comment:
Use Map interface here.
```suggestion
    private final Map<T, FieldValueHitQueue<FieldValueHitQueue.Entry>[]> groupQueueMap;
```

Reply:
Similarly wherever possible.